.gitlab-ci.yml

```diff
@@ -0,0 +1,147 @@
+---
+# To contribute improvements to CI/CD templates, please follow the Development guide at:
+# https://docs.gitlab.com/ee/development/cicd/templates.html
+# This specific template is located at:
+# https://gitlab.com/gitlab-org/gitlab/-/blob/master/lib/gitlab/ci/templates/Python.gitlab-ci.yml
+include:
+  - project: nci-gdc/gitlab-templates
+    ref: master
+    file:
+      - templates/global/full.yaml
+      - templates/python/full.yaml
+
+variables:
+ BASE_CONTAINER_VERSION: "1.4.0"
+ PRE_COMMIT_DOCKER_IMAGE: docker.osdc.io/ncigdc/python3.8-builder:2.1.0
+ DOCKER_BUILDKIT: 1
+ PIP_EXTRA_INDEX_URL: https://nexus.osdc.io/repository/pypi-all/simple
+ DOCKER_BUILD_OPTS: "--build-arg PIP_EXTRA_INDEX_URL=https://nexus.osdc.io/repository/pypi-all/simple"
+ DOCKER_PUSH_OPTS: "--all-tags"
+ TWINE_REPOSITORY_URL: "https://nexus.osdc.io/repository/pypi-snapshots/"
+
+
+tox:
+  image: ${BASE_CONTAINER_REGISTRY}/${REPO_PY_VERSION}-builder:${BASE_CONTAINER_VERSION}
+  before_script:
+    - mkdir -p /usr/share/man/man1
+
+.python_versions:
+  parallel:
+    matrix:
+      - REPO_PY_VERSION: ["python36"]
+
+docker_build:
+  allow_failure: true
+  rules:
+    - when: never
+
+release:
+  image: docker.osdc.io/ncigdc/python36-builder:1.4.0
+  allow_failure: true
+  rules:
+    - when: never
+
+update_reqs_for_gdcdatamodel2:
```

Comment:
I think this should use the `needs` option to depend on the main gdcdictionary tests completing successfully before triggering gdcdatamodel2 work. And can we rename the stage so its clearly different from the build step of gdcdictionary?

Reply:
>I think this should use the needs option to depend on the main gdcdictionary tests completing successfully before triggering gdcdatamodel2 work
<img width="1282" alt="Screen Shot 2023-02-14 at 2 05 58 PM" src="https://user-images.githubusercontent.com/68757394/218849769-65fe6c44-f09f-486b-80ea-130fa9e7e977.png">
It already does that. The later stage usually does not start until the earlier stage finish. The only exception is `code_quality` not sure why.

Reply:
How is secret_detection different than the pre-commit step? Is it duplicating effort?

Reply:
I think it is, it is gitlab's implementation.

https://docs.gitlab.com/ee/user/application_security/secret_detection/

Reply:
>And can we rename the stage so its clearly different from the build step of gdcdictionary?

<img width="1218" alt="Screen Shot 2023-02-14 at 3 39 29 PM" src="https://user-images.githubusercontent.com/68757394/218869139-2e1a818a-3b28-4492-9510-7bf9ac4c6b5f.png">

Reply:
I think the difference is you want this stage to run only when the previous stage passes, not sure if that is handled automatically. For example, we don't want to build gdcdatamodel2 when gdcdictionary test is failing

Reply:
<img width="1292" alt="Screen Shot 2023-02-14 at 4 18 46 PM" src="https://user-images.githubusercontent.com/68757394/218876093-2a79caad-d201-4e6e-b305-b318fb39bac6.png">

Reply:
I think the later stage won't run if a job in the previous stage failed.